install-dev/upgrade/php/ps_1770_update_charset.php

```diff
@@ -0,0 +1,91 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+function ps_1770_update_charset()
+{
+    $adminFilterTableExists = $adminFilterFilterIdExists = $moduleHistoryTableExists = $translationTableExists = false;
+
+    try {
+        $adminFilterTableExists = (bool) Db::getInstance()->executeS(
+            'SELECT count(*) FROM ' . _DB_PREFIX_ . 'admin_filter'
+        );
+        if ($adminFilterTableExists) {
+            $adminFilterFilterIdExists = (bool) Db::getInstance()->executeS(
+                'SELECT count(filter_id) FROM ' . _DB_PREFIX_ . 'admin_filter'
+            );
+        }
+    } catch (Exception $e) {
+
+    }
+
+    try {
+        $moduleHistoryTableExists = (bool) Db::getInstance()->executeS(
```

Comment:
I'm not sure but what if the code fail? is it possible that the `$moduleHistoryTableExists` variable will not be initialized? The the following code could fail because of an undefined variable

So maybe in the next code you should check `if (isset($moduleHistoryTableExists) && $moduleHistoryTableExists)` Or define the variable `$moduleHistoryTableExists = false;` before the try/catch block?

Reply:
It's defined before: https://github.com/PrestaShop/PrestaShop/pull/22274/files#diff-e02480de244d3b5fdbd2e8354f049d52b6602b5ec711b7608d2e0dbf4bea0929R29